docs/development/workflow/development_workflow.rst

```diff
@@ -460,17 +479,62 @@ You are more likely to run into *conflicts* here--places where the changes you
 made conflict with changes that someone else made--than anywhere else. Ask for
 help if you need it.
 
-After the rebase you need to push your changes to GitHub; you will need force
-the push because `git`_ objects to re-writing the history of the repository
-after you have pushed it somewhere::
+.. _howto_squash:
+
+How to squash
+=============
+
+Typically we ask to *squash* when there was a fair amount of trial
+and error, but the final patch remains quite small, or when files were added
+and removed (especially binary files or files that should not remain in the
+repository) or if the number of commits in the history is disproportionate
+compared to the work being carried out (for example 30 commits gradually
+refining a final 10-line change).  Conceptually this is equivalent to
+exporting the final diff from a feature branch, then starting a new branch and
+applying only that patch.
+
+The actual squashing is done on a new branch, using ``git merge --squash``::
```

Comment:
@mhvk: pushed.

_(It's probably quickly if others could push any ideas/improves to the PR branch directly).